### PR TITLE
[BACK-54] Second Version of dbUpdater

### PR DESCRIPTION
**Pull request by thomas-woehrle**
### Description of Issue
Improvement of the first version. 

This includes:
- distinction between users or organizations as owners
- integration of star-history and languages used in a project
- integration of updateRepos function
- adjusting the db correspondingly
### How I implemented
- restructured the calling of functions
- added` is_trending_daily, is_trending_weekly, is_trending_monthly` on db 
- these columns are purged and updated daily
- also added ELI5 integration of Maxis and Alex' functions
### Other
- right now updateProject and insertProject do very similar things but later on hackernewsPosts for example will not be updated on a daily basis

### CODE REVIEW

File: packages/graphql_server/package.json [hunk 1]
```diff
@@ -11,9 +11,11 @@
   "scripts": {
     "dev": "concurrently -c 'magenta' 'tsc -w' '$npm_package_config_env_dev && nodemon dist/index.js'",
     "build": "tsc",
-    "serve": "$npm_package_config_env_dev && node dist/index.js"
+    "serve": "$npm_package_config_env_dev && node dist/index.js",
+    "update-types": "npx supabase gen types typescript --project-id \"yipnhkmklmbuxjwzpipg\" > types/supabase.d.ts"
   },
   "dependencies": {
+    "@supabase/supabase-js": "^2.23.0",
```
Comment (pczern) [inline packages/graphql_server/package.json:18]: if this dependency is added here, then remove the supabase dependency, delete package-lock.json/node_modules and run npm i again

Comment (pczern) [reply packages/graphql_server/package.json:18]: we will use supabase-js dependency to fix possible former deployment bug

File: packages/graphql_server/package.json [hunk 2]
```diff
@@ -11,9 +11,11 @@
   "scripts": {
     "dev": "concurrently -c 'magenta' 'tsc -w' '$npm_package_config_env_dev && nodemon dist/index.js'",
     "build": "tsc",
-    "serve": "$npm_package_config_env_dev && node dist/index.js"
+    "serve": "$npm_package_config_env_dev && node dist/index.js",
+    "update-types": "npx supabase gen types typescript --project-id \"yipnhkmklmbuxjwzpipg\" > types/supabase.d.ts"
```
[the remark this answers is not in the document]
Comment (pczern) [reply packages/graphql_server/package.json:15]: get project-id from env variables

File: packages/graphql_server/src/dataAggregation.ts [hunk 3]
```diff
@@ -129,3 +137,73 @@ const getOrganizationID = async (owner: string) => {
...
+      .insert([personDataDBFormat])
+    personInsertionError &&
+      console.error('Error inserting organization into database: \n', personInsertionError)
+
+    const { data: person } = await supabase
+      .from('associated_person')
+      .select('id')
+      .eq('login', owner)
+
+    if (person?.[0]?.id) {
```
Comment (pczern) [inline packages/graphql_server/src/dataAggregation.ts:203]: ternary expression

**Review by pczern — CHANGES_REQUESTED** [review]

File: packages/graphql_server/src/processRepo.ts [hunk 4]
```diff
@@ -1,72 +1,145 @@
...
+  )
+  // convert the github data into a format that can be inserted into the database. ProjectUpdate and ProjectInsertion are virtually  the same
+  const projectUpdate: ProjectUpdate = turnIntoProjectInsertion(
+    githubData,
+    starHistory
+  ) as ProjectUpdate
+
+  const updated = await updateSupabaseProject(name, owner, projectUpdate)
+
+  updated ? console.log('updated ', name, 'owned by', owner) : null
```
[the remark this answers is not in the document]
Comment (thomas-woehrle) [reply packages/graphql_server/src/processRepo.ts:73]: Would prefer to keep. We will move to a logger later on, but for now it is nice to see what is happening when running this function.

File: packages/graphql_server/types/dataAggregation.d.ts [hunk 5]
```diff
@@ -2,3 +2,5 @@ import { Database } from './supabase'
 
 export type ProjectInsertion = Database['public']['Tables']['project']['Insert']
 export type OrganizationInsertion = Database['public']['Tables']['organization']['Insert']
+export type UserInsertion = Database['public']['Tables']['user']['Insert']
```
[the remark this answers is not in the document]
Comment (thomas-woehrle) [reply packages/graphql_server/types/dataAggregation.d.ts:5]: When i do it this way and hover over a type of ProjectInsertion i get an odd description of what the type looks like. So would prefer to keep like this.

**Comment by thomas-woehrle** [comment]
@pczern @cnk1dev 
I addressed your change requests

File: packages/graphql_server/src/dataAggregation.ts [hunk 6]
```diff
@@ -177,33 +190,24 @@ export const getPersonID = async (owner: string) => {
...
+
+  const { error: personInsertionError } = await supabase
+    .from('associated_person')
+    .insert([personDataDBFormat])
+  personInsertionError &&
+    console.error('Error inserting organization into database: \n', personInsertionError)
+
+  const { data: person } = await supabase.from('associated_person').select('id').eq('login', owner)
+
+  return existing_person?.[0]?.id ? existing_person[0].id : null
```
Comment (cnk1dev) [inline packages/graphql_server/src/dataAggregation.ts:212]: Did you mean to use person instead of existing_person here?

**Review by cnk1dev — APPROVED** [review]
Double-check the return statement in the comment below, otherwise good to go from my side!

File: packages/graphql_server/types/dataAggregation.d.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (pczern) [reply packages/graphql_server/types/dataAggregation.d.ts:5]: if there is no error, change it please; you are not required to hover

File: packages/graphql_server/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (pczern) [reply packages/graphql_server/package.json:15]: if you us `$npm_package_config_env_dev && <your command>` and then get the environment variable like $SUPABASE_REFERENCE_ID it will work

[the remark this answers is not in the document]
Comment (thomas-woehrle) [reply packages/graphql_server/package.json]: thanks, changed it

**Comment by thomas-woehrle** [comment]
I will not merge until BACK-56 is merged to check for conflicts

**Comment by thomas-woehrle** [comment]
Never mind might be better to merge this one first
